### PR TITLE
feat: add docker-compose-language-service

### PR DESCRIPTION
**Pull request by marcdeop**
Installs [docker-compose-language-service](https://github.com/microsoft/compose-language-service)

### CODE REVIEW

File: lua/mason-registry/docker-compose-language-service/init.lua [hunk 1]
```diff
@@ -0,0 +1,12 @@
+local Pkg = require "mason-core.package"
+local npm = require "mason-core.managers.npm"
+
+return Pkg.new {
+    name = "docker-compose-language-service",
+    desc = [[A language server for Docker Compose.]],
+    homepage = "https://github.com/microsoft/compose-language-service",
+    languages = { Pkg.Lang.YAML },
+    categories = { Pkg.Cat.LSP },
+    install = npm.packages { "@microsoft/compose-language-service" },
```
Comment (williamboman) [inline lua/mason-registry/docker-compose-language-service/init.lua:10]: They don't seem to be providing an executable to launch the server. I think we should provide one via Mason - or is there some reason there's no executable provided?

Comment (marcdeop) [reply lua/mason-registry/docker-compose-language-service/init.lua:10]: You are right, they don't provide it.

I figured I started with just the installation and try adding the executable afterwards.

However, I am struggling to find a nice way of doing so :S

**Comment by williamboman** [comment]
@williambotman /apply

```diff
diff --git a/lua/mason-registry/docker-compose-language-service/init.lua b/lua/mason-registry/docker-compose-language-service/init.lua
index 58ed76d..1a8d2c0 100644
--- a/lua/mason-registry/docker-compose-language-service/init.lua
+++ b/lua/mason-registry/docker-compose-language-service/init.lua
@@ -1,12 +1,29 @@
 local Pkg = require "mason-core.package"
+local path = require "mason-core.path"
 local npm = require "mason-core.managers.npm"
 
 return Pkg.new {
     name = "docker-compose-language-service",
     desc = [[A language server for Docker Compose.]],
     homepage = "https://github.com/microsoft/compose-language-service",
-    languages = { Pkg.Lang.YAML },
+    languages = { Pkg.Lang.Docker },
     categories = { Pkg.Cat.LSP },
-    install = npm.packages { "@microsoft/compose-language-service" },
+    ---@async
+    ---@param ctx InstallContext
+    install = function(ctx)
+        npm.install({ "@microsoft/compose-language-service" }).with_receipt()
+        ctx:link_bin(
+            "docker-compose-language-service",
+            ctx:write_node_exec_wrapper(
+                "docker-compose-language-service",
+                path.concat {
+                    "node_modules",
+                    "@microsoft",
+                    "compose-language-service",
+                    "lib",
+                    "server.js",
+                }
+            )
+        )
+    end,
 }
-
diff --git a/lua/mason-registry/dockerfile-language-server/init.lua b/lua/mason-registry/dockerfile-language-server/init.lua
index 71eada1..e1eb6df 100644
--- a/lua/mason-registry/dockerfile-language-server/init.lua
+++ b/lua/mason-registry/dockerfile-language-server/init.lua
@@ -5,7 +5,7 @@ return Pkg.new {
     name = "dockerfile-language-server",
     desc = [[A language server for Dockerfiles powered by Node.js, TypeScript, and VSCode technologies.]],
     homepage = "https://github.com/rcjsuen/dockerfile-language-server-nodejs",
-    languages = { Pkg.Lang.Dockerfile },
+    languages = { Pkg.Lang.Docker },
     categories = { Pkg.Cat.LSP },
     install = npm.packages { "dockerfile-language-server-nodejs", bin = { "docker-langserver" } },
 }
diff --git a/lua/mason-registry/hadolint/init.lua b/lua/mason-registry/hadolint/init.lua
index b634213..55a7787 100644
--- a/lua/mason-registry/hadolint/init.lua
+++ b/lua/mason-registry/hadolint/init.lua
@@ -10,7 +10,7 @@ return Pkg.new {
     name = "hadolint",
     desc = [[Dockerfile linter, validate inline bash, written in Haskell]],
     homepage = "https://github.com/hadolint/hadolint",
-    languages = { Pkg.Lang.Dockerfile },
+    languages = { Pkg.Lang.Docker },
     categories = { Pkg.Cat.Linter },
     ---@async
     ---@param ctx InstallContext
```

**Comment by williamboman** [comment]
@williambotman /fixup

**Comment by OGrigorios** [comment]
Sorry for bumping here, but I am confused on how to make this language-service work for docker-compose files. Can anyone of you saints provide me with a configuration schema?

**Comment by marcdeop** [comment]
> Sorry for bumping here, but I am confused on how to make this language-service work for docker-compose files. Can anyone of you saints provide me with a configuration schema?

Please, do not abuse this to get support.

Raise your questions through the appropriate channels.

**Pull request merged**